### PR TITLE
Picopass: detect and show SE / SIO

### PR DESCRIPTION
**Pull request by pcunning**
# What's new

- Adds detection and display of SE/SR cards.

# Verification 

- Install with picopass app
- scan SE or SR card

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

cc: @bettse

### CODE REVIEW

**Comment by me0wday** [comment]
I was testing this out, but it seems to cause a hardfault on my flipper, the following is all that is shown in logs:
```
68149 [D][PicopassWorker] rfalPicoPassPollerReadBlock block 70
68160 [D][PicopassWorker] rfalPicoPassPollerReadBlock 70 a30303030003e017
68162 [D][PicopassWorker] rfalPicoPassPollerReadBlock block 71
68189 [E][PicopassWorker] rfalPicoPassPollerReadBlock error 4
68191 [E][PicopassWorker] picopass_read_card error 4
68194 [E][PicopassWorker] picopass_device_parse_credential error 4
68198 [E][PicopassWorker] picopass_device_parse_wiegand error 4
```

**Comment by bettse** [comment]
Is that the beginning of the PicopassWorker logs?  Do you see any other block data?  Specifically the "aia" and "config" would be useful for getting more context.  The data you see on block 70 is on block 6 for me.  Is this a card (vs fob)?  Can you tell us what it says after "HID iClass" on the bottom left side of the back of the card?

**Comment by me0wday** [comment]
> Is that the beginning of the PicopassWorker logs? Do you see any other block data? Specifically the "aia" and "config" would be useful for getting more context. The data you see on block 70 is on block 6 for me. Is this a card (vs fob)? Can you tell us what it says after "HID iClass" on the bottom left side of the back of the card?

Sure, here is the start:
It's a white card with the text "HID iCLASS"
```
67287 [D][PicopassWorker] config 0000000000000000
67291 [T][StorageAPI] File 0000C9D0 - 0000CD44 open (/int/.power.settings)
67300 [T][StorageAPI] File 0000C9D0 - 0000CD44 closed
67303 [T][StorageAPI] File/Dir 0000C9D0 free
67307 [D][PicopassWorker] aia ffffffffffffffff
67328 [D][PicopassWorker] rfalPicoPassPollerReadBlock block 2
67337 [D][PicopassWorker] rfalPicoPassPollerReadBlock 2 eeffffffffffffff
67341 [D][PicopassWorker] rfalPicoPassPollerReadBlock block 3
67351 [D][PicopassWorker] rfalPicoPassPollerReadBlock 3 ffffffffffffffff
67353 [D][PicopassWorker] rfalPicoPassPollerReadBlock block 4
67364 [D][PicopassWorker] rfalPicoPassPollerReadBlock 4 ffffffffffffffff
67367 [D][PicopassWorker] rfalPicoPassPollerReadBlock block 5
67377 [D][PicopassWorker] rfalPicoPassPollerReadBlock 5 ffffffffffffffff
67379 [D][PicopassWorker] rfalPicoPassPollerReadBlock block 6
67389 [D][PicopassWorker] rfalPicoPassPollerReadBlock 6 a30303030003e017
```

**Comment by bettse** [comment]
>  It's a white card with the text "HID iCLASS"

Anything after that?  

Does this happen consistently?  The config being all 00 looks wrong, and I wonder at first if it was a bad read.

File: applications/picopass/picopass_worker.c [hunk 1]
```diff
@@ -152,15 +209,9 @@ ReturnCode picopass_read_card(PicopassBlock* AA1) {
...
-    err = rfalPicoPassPollerReadBlock(PICOPASS_CSN_BLOCK_INDEX, &csn);
-    memcpy(AA1[PICOPASS_CSN_BLOCK_INDEX].data, csn.data, sizeof(csn.data));
-
-    rfalPicoPassReadBlockRes cfg;
-    err = rfalPicoPassPollerReadBlock(PICOPASS_CONFIG_BLOCK_INDEX, &cfg);
-    memcpy(AA1[PICOPASS_CONFIG_BLOCK_INDEX].data, cfg.data, sizeof(cfg.data));
-
-    size_t app_limit = cfg.data[0] < PICOPASS_MAX_APP_LIMIT ? cfg.data[0] : PICOPASS_MAX_APP_LIMIT;
+    size_t app_limit = AA1[PICOPASS_CONFIG_BLOCK_INDEX].data[0] < PICOPASS_MAX_APP_LIMIT ?
+                           AA1[PICOPASS_CSN_BLOCK_INDEX].data[0] :
```
Comment (bettse) [inline applications/picopass/picopass_worker.c]: oh shit.  This says `PICOPASS_CSN_BLOCK_INDEX` not `PICOPASS_CONFIG_BLOCK_INDEX`

**Comment by pcunning** [comment]
@me0wday give the update a try and let us know if it's fixed.

**Comment by bettse** [comment]
LGTM!

**Comment by FalsePhilosopher** [comment]
HID iClass DP, it now says 255 bits and no info an when you try to load the saved file it says can not parse file. I have the same card saved without this PR and it says 37 bits and the 10 chars. Also how are you getting those logs? I'm using cli logs but nothing picopass related shows up.

**Comment by pcunning** [comment]
Would you be able to share your file and the debug logs? If not publicly feel free to email or DM on discord either @bettse or myself. 

For logs set the log level to debug in settings and open the cli before reading.

**Comment by FalsePhilosopher** [comment]
> Would you be able to share your file and the debug logs? If not publicly feel free to email or DM on discord either @bettse or myself.
> 
> For logs set the log level to debug in settings and open the cli before reading.

Email sent with both files.
using this PR
105367 [D][PicopassWorker] csn f0ce2c15feff12e0
105393 [D][PicopassWorker] config 0000000000000000
105403 [D][PicopassWorker] aia ffffffffffffffff
105418 [D][PicopassDevice] Unknown encryption
105421 [D][PicopassDevice] bitLength: 255

Email sent with a log of working prePR card read.

**Comment by pcunning** [comment]
Thank you for the data that should help! Looks like I might have a bug with the config data to work out.

**Comment by pcunning** [comment]
@FalsePhilosopher I think that should fix the issues you were seeing and also fix the root of the issue @me0wday was having.

**Comment by skotopes** [comment]
@pcunning may I ask you to rebase?

**Comment by pcunning** [comment]
@skotopes updated and working for me

**Review by skotopes — APPROVED** [review]

**Comment by me0wday** [comment]
Sorry guys, been a while since I've had access to my Flipper, just gave this PR another try and it's working :)

**Pull request merged**